addons/irlight/XEH_postInit.sqf

```diff
@@ -1,30 +1,33 @@
 #include "script_component.hpp"
 
-[] call FUNC(initItemContextMenu);
+call FUNC(initItemContextMenu);
 
-addUserActionEventHandler ["headlights", "Deactivate", LINKFUNC(onLightToggled)];
+//addUserActionEventHandler ["headlights", "Activate", {LINKFUNC(onLightToggled) call CBA_fnc_execNextFrame}];
 
 ["ACE3 Equipment", QGVAR(hold), LLSTRING(MomentarySwitch), {
-    ACE_player action ["GunLightOn", ACE_player];
-    ACE_player action ["IRLaserOn", ACE_player];
-    [] call FUNC(onLightToggled);
+    if !(ACE_player call CBA_fnc_canUseWeapon) exitWith {};
+
+    // Save current weapon state to reapply later
+    private _weaponState = (weaponState ACE_player) select [0, 3];
+
+    action ["GunLightOn", ACE_player];
+    action ["IRLaserOn", ACE_player];
+
+    ACE_player selectWeapon _weaponState;
+
+    // call FUNC(onLightToggled);
+
     true
 }, {
-    ACE_player action ["GunLightOff", ACE_player];
-    ACE_player action ["IRLaserOff", ACE_player];
-    [] call FUNC(onLightToggled);
-    true
-}] call CBA_fnc_addKeybind;
+    if !(ACE_player call CBA_fnc_canUseWeapon) exitWith {};
+
+    // Save current weapon state to reapply later
+    private _weaponState = (weaponState ACE_player) select [0, 3];
+
+    action ["GunLightOff", ACE_player];
+    action ["IRLaserOff", ACE_player];
 
-["CBA_attachmentSwitched", {
-    params ["", "", "_item"];
-
-    private _substr = _item select [0, 8];
-    if (
-        ACE_player getVariable [QGVAR(isTurnedOn), false]
-        && {_substr == "ACE_SPIR" || {_substr == "ACE_DBAL"}}
-    ) then {
-        ACE_player action ["GunLightOn", ACE_player];
```

Comment:
I think we still want this
before you could cycle modes (ctrl-L) and it would keep the light on